src/test/java/org/opensearch/knn/indices/ModelGraveyardTests.java

```diff
@@ -57,6 +60,24 @@ public void testStreams() throws IOException {
         assertTrue(testModelGraveyardCopy.contains(testModelId));
     }
 
+    public void testXContentBuilder() throws IOException {
+        Set<String> modelIds = new HashSet<>();
+        String testModelId = "test-model-id";
+        String testModelId1 = "test-model-id1";
+        modelIds.add(testModelId);
+        modelIds.add(testModelId1);
+        ModelGraveyard testModelGraveyard = new ModelGraveyard(modelIds);
+
+        XContentBuilder xContentBuilder = XContentFactory.jsonBuilder();
+        xContentBuilder.startObject();
+        XContentBuilder builder = testModelGraveyard.toXContent(xContentBuilder, ToXContent.EMPTY_PARAMS);
+        builder.endObject();
+
+        ModelGraveyard testModelGraveyard2 = ModelGraveyard.fromXContent(createParser(builder));
```

Comment:
Validate as well that there are only 2 entries in modelIds.

Reply:
updated